libraries/chain/apply_context.cpp

```diff
@@ -64,8 +64,11 @@ void apply_context::exec_one()
          kv_iterators.resize(1);
          kv_destroyed_iterators.clear();
          if (!context_free) {
-            kv_ram = create_kv_chainbase_context(db, kvram_id, receiver, create_kv_resource_manager_ram(*this), control.get_global_properties().kv_configuration.kvram);
-            kv_disk = create_kv_chainbase_context(db, kvdisk_id, receiver, create_kv_resource_manager_disk(*this), control.get_global_properties().kv_configuration.kvdisk);
+            auto get_deep_mind_logger = [this]() { return control.get_deep_mind_logger(); };
+            auto get_action_id = [this]() { return this->get_action_id(); };
```

Comment:
The `kv_resource_manager` already has access to the current `apply_context`. So you can get access to both `get_action_id()` and `control.get_deep_mind_logger` from there rather than having to pass in these two functions.

Reply:
Those are actually used by `kv_context_chainbase`. It only has indirect access to the context if using `this.resource_manager._context.get...`. Since it was starting with an underscore I thought I could not use it.

So you are ok if I use `resource_manager._context` inside the `kv_context_chainbase` instance?

Reply:
Usage looks like this: `auto dm_logger = resource_manager._context->control.get_deep_mind_logger()`.

Should I create a getter in `apply_context` or I leave this as-is.

Reply:
> So you are ok if I use resource_manager._context inside the kv_context_chainbase instance?

Yes.

And adding a getter for convenience is fine.

Reply:
I decided on not adding the getter in the end. Did not feel it worth it.